### PR TITLE
Recreate ghaf-coverity host

### PR DESCRIPTION
**Pull request by fayadf**
* add new host configurations
* package binaries from tii licensed host
* make derivation for build with deps
* bring derivation in pkgs

### CODE REVIEW

File: hosts/ghaf-coverity/packages.nix
Comment (joinemm) [inline hosts/ghaf-coverity/packages.nix:1]: I would probably just merge this file with the configuration.nix, there's no need to separate a short package list.

File: pkgs/coverity/default.nix [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  lib,
+  autoPatchelfHook,
+  fetchurl,
+  curl,
+  systemd,
+  zlib,
+  xorg,
+  alsaLib,
+  libxcrypt-legacy,
+  withUpdater ? true,
```
Comment (joinemm) [inline pkgs/coverity/default.nix:14]: I think we have no use for the `withUpdater` conditional

File: pkgs/coverity/default.nix [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+  nativeBuildInputs = [ autoPatchelfHook ];
+  buildInputs =
+    [
+      # libudev
+      (lib.getLib systemd)
+      # libstdc++.so libgcc_s.so
+      stdenv.cc.cc.lib
+    ]
+    ++ lib.optionals withUpdater [
```
Comment (joinemm) [inline pkgs/coverity/default.nix:34]: Package list can be flattened when the conditional is removed

**Review by joinemm — CHANGES_REQUESTED** [review]

File: hosts/ghaf-coverity/packages.nix
Comment (fayadf) [reply hosts/ghaf-coverity/packages.nix:1]: Thanks, cleaned up.

File: pkgs/coverity/default.nix (hunk 1 above)
Comment (fayadf) [reply pkgs/coverity/default.nix]: Yes, removed and fixed.

File: pkgs/coverity/default.nix (hunk 2 above)
Comment (fayadf) [reply pkgs/coverity/default.nix]: Thanks, it is organized now.

File: hosts/ghaf-coverity/secrets.yaml [hunk 3]
```diff
@@ -1,30 +1,39 @@
...
+ssh_host_ed25519_key: ENC[AES256_GCM,data:mz/nbtkpCCI1IMqXMne6/i5F6JwpXvuNCs9DbNCgQGYP2MtVYAIL0FLi4glpWmE64rPeF60XLoNFGRVomALwF6we+a8YbA5y1rNx/Zympn/1RJ7GqMoEhJ4vHOWUyYeTWnzD0NGfLdmcBvTaWRT9a3jFLj9plYD5DnNluCdBzzif0XCBSlAnM3KAlJVY/0kByAo1aXCjQ1p67xN8yRuSbrJ6+DD+Q2cL8ITuUGQROZa9c06aWahwfmzESscfm623T3k6t9IiePojHI6wLNZJ2wKHjtvtRZjhLt6ykYSMwfxdZTf3xxvGW88z24hHCs1fpMBowIh2z5ip1kAsZcWpohjZL/QKMSjW3/zAR99qkG/2BDmK0lrEs50plWh4B21hEFf7iHAT049qTUf219P6iPlyMVW4B1l3C2eYsWZQQsWSwamzAw2h94RFQtoEdfcZVg7M0fmlO4Q2smHn5YsqxkxMghVSyLb+ytcl29bq2AnKBZt4gkzCTeSYzdnL0wfZyaYd5ITHRJNz3eJFlXrq1XuQj8wZbr4Hj9r9,iv:dlg6VLNBUgD7I9dsalaDkZSwaacPDkd8GraM/xBPTow=,tag:/7V8JCGBmForB2KUz+Aw9w==,type:str]
 sops:
     kms: []
     gcp_kms: []
     azure_kv: []
     hc_vault: []
     age:
-        - recipient: age172azvwv5vne79mqfhvdvk9j95gn5v04uk9t3fjdfe5p7dv7kucvqpygxkx
+        - recipient: age1z825k99myjmfcml86pujcmtj96psvj8c3m08me8kkq03tkpwy9xql4jt9y
```
Comment (joinemm) [inline hosts/ghaf-coverity/secrets.yaml:8]: New host age key should also be updated in `.sops.yaml`

Comment (fayadf) [reply hosts/ghaf-coverity/secrets.yaml:8]: Thanks. Updated the file.

**Review by joinemm — APPROVED** [review]

**Review by joinemm — APPROVED** [review]

**Pull request merged**